org.gecko.emf.exporter/src/org/gecko/emf/exporter/AbstractEMFExporter.java

```diff
@@ -0,0 +1,1766 @@
+import java.util.stream.Collectors;
+
+import org.eclipse.emf.ecore.EAnnotation;
+import org.eclipse.emf.ecore.EAttribute;
+import org.eclipse.emf.ecore.EClass;
+import org.eclipse.emf.ecore.EDataType;
+import org.eclipse.emf.ecore.EEnum;
+import org.eclipse.emf.ecore.EEnumLiteral;
+import org.eclipse.emf.ecore.EObject;
+import org.eclipse.emf.ecore.EReference;
+import org.eclipse.emf.ecore.EStructuralFeature;
+import org.eclipse.emf.ecore.resource.Resource;
+import org.eclipse.emf.ecore.util.EcoreUtil;
+import org.gecko.emf.exporter.cells.EMFExportEObjectIDValueCell;
+import org.gecko.emf.exporter.cells.EMFExportEObjectManyReferencesValueCell;
+import org.gecko.emf.exporter.cells.EMFExportEObjectOneReferenceValueCell;
+import org.gecko.emf.exporter.cells.EMFExportMappingMatrixReferenceValueCell;
+import org.gecko.emf.exporter.headers.EMFExportEObjectColumnHeader;
+import org.gecko.emf.exporter.headers.EMFExportEObjectGenericColumnHeader;
+import org.gecko.emf.exporter.headers.EMFExportEObjectIDColumnHeader;
+import org.gecko.emf.exporter.headers.EMFExportEObjectManyReferencesColumnHeader;
+import org.gecko.emf.exporter.headers.EMFExportEObjectOneReferenceColumnHeader;
+import org.gecko.emf.exporter.keys.EMFExportRefMatrixNameIDCompositeKey;
+import org.slf4j.Logger;
+
+import com.google.common.base.Stopwatch;
+import com.google.common.collect.HashBasedTable;
+import com.google.common.collect.Table;
+
+/**
+ * Implementation of format-agnostic methods for exporting EMF resources and
+ * lists of EMF objects to desired format.
+ * 
+ * Format-specific implementations should extend this class and only focus on
+ * transformation of matrices resulting from {@link #exportEObjectsToMatrices}
+ * to arrive at desired format.
+ * 
+ * @author Michal H. Siemaszko
+ */
+public abstract class AbstractEMFExporter implements EMFExporter {
+	protected static final List<String> METADATA_ECLASS_MATRIX_COLUMNS_HEADERS = List.of("Name", "Type", "isMany",
+			"isRequired", "Default value", "Documentation");
+
+	protected static final List<String> METADATA_EENUM_MATRIX_COLUMNS_HEADERS = List.of("Name", "Literal", "Value",
+			"Documentation");
+
+	protected static final String METADATA_DOCUMENTATION_HEADER = "Documentation";
+	protected static final String METADATA_PSEUDOID_DOCUMENTATION = "No default ID present, therefore pseudo-ID was generated for identification purposes";
+
+	protected static final String METADATA_MATRIX_NAME_SUFFIX = "Metadata";
+	protected static final String MAPPING_MATRIX_NAME_SUFFIX = "Mapping";
+
+	protected static final String DOCUMENTATION_GENMODEL_SOURCE = "http://www.eclipse.org/emf/2002/GenModel";
+	protected static final String DOCUMENTATION_GENMODEL_DETAILS = "documentation";
+
+	protected static final String ECORE_PACKAGE_NAME = "ecore";
+
+	protected static final String ID_COLUMN_NAME = "Id";
+
+	protected static final String ID_ATTRIBUTE_NAME = "id";
+
+	protected static final String REF_COLUMN_PREFIX = "ref_";
```

Comment:
Please change this to a sufix. A Ref should become e.g.`family.father._ref`. This would be closer to e.g. mongoEMF (`family.father._eProxyUri`) and EMFJson (`family.father.$ref`). With this we also establish the pattern, that _ marks a reserved property, that is not part of the actual model.